### PR TITLE
new option to output all janno columns in list

### PR DESCRIPTION
**Pull request by stschiff**
This PR introduces an API-flag `--fullJanno` for retrieving extended individual info with all standard janno columns. A similar option exists for the Web API, see Changelog.

### CODE REVIEW

File: src/Poseidon/CLI/OptparseApplicativeParsers.hs [hunk 1]
```diff
@@ -600,7 +600,8 @@ parseListEntity = parseListPackages <|> parseListGroups <|> (parseListIndividual
     parseListIndividualsDummy = OP.flag' () (
         OP.long "individuals" <>
         OP.help "List all individuals/samples.")
-    parseListIndividualsExtraCols = ListIndividuals <$> OP.many parseExtraCol
+    parseListIndividualsExtraCols = ListIndividuals <$> (parseAllJannoCols <|> (Just <$> OP.many parseExtraCol))
+    parseAllJannoCols = OP.flag' Nothing (OP.long "fullJanno" <> OP.help "output all Janno Columns")
```
Comment (nevrome) [inline src/Poseidon/CLI/OptparseApplicativeParsers.hs:604]: It's a bit inconsistent that for the Web-API we have a keyword to express "give me all" (`/individuals?additionalJannoColumns=ALL`), but on the command line we have a special option (`--fullJanno`).

We could also use a keyword on the command line (`--jannoColumn ALL`). Or on the API side a special option (`/individuals?fullJanno`).

But CLI and API are already inconsistent in that `--jannoColumn` takes only one column name, but can be listed multiple times, whereas `additionalJannoColumns` of the webserver takes a list of column names.

Up to you - just wanted to point it out.

**Review by nevrome — COMMENTED** [review]
I tested to query the current webserver with `--fullJanno`. Of course it just gives back an empty column `ALL`:

image

Maybe this is not ideal. And it could probably be avoided with an extra `?fullJanno` API option instead of `&additionalJannoColumns=ALL` (see my comment above). With this solution the old server would just ignore the additional API option, which is better (But also not perfect. Ideally the server should report unknown flags with a warning, right?).

File: src/Poseidon/CLI/List.hs [hunk 2]
```diff
@@ -30,15 +30,15 @@ data ListOptions = ListOptions
...
+    , _listOnlyLatest   :: Bool -- ^ whether to show only latest versions of packages
     }
 
 data RepoLocationSpec = RepoLocal [FilePath] | RepoRemote ArchiveEndpoint
 
 -- | A datatype to represent the options what to list
 data ListEntity = ListPackages
     | ListGroups
-    | ListIndividuals [String]
+    | ListIndividuals (Maybe [String]) -- Nothing means all Janno columns. Just [] means none.
```
[the remark this answers is not in the document]
Comment (stschiff) [reply src/Poseidon/CLI/List.hs:41]: Yeah OK. In this case I tend to agree. I usually avoid Boilerplate types if they are reducible to `Maybe` or `Either`, but in this case you may be right. I can do that.

File: src/Poseidon/CLI/OptparseApplicativeParsers.hs (hunk 1 above)
Comment (stschiff) [reply src/Poseidon/CLI/OptparseApplicativeParsers.hs:604]: I will reply in the main conversation thread.

**Comment by stschiff** [comment]
Thanks for the review @nevrome, I will work on it, see my response above. Regarding the Web-API, I don't quite understand why we would like to cater for the case of using a new client with an old server? That should never happen, unless users actually use `trident serve` locally, which would surprise me at this point. Our server will always be up to date. So I don't see how the behaviour you discovered above would have any ramifications in production?

With respect to the `additionalJannoColumns=ALL` -> I get your point, but I think introducing an additional option is also not great, because it would lead to illegal semantics, like: `&additionalJannoColumns=Country,Publication&fullJanno`. Also, I am not even sure whether URLs even support boolean-like switch options like the one here. The usual way options are used is as key-value pair.

I think the rule that `ALL` retains a special meaning as a column name is a good option considering all alternatives. It is simple, easy to remember and avoids illegal option-combinations. What do you think?

That the CLI introduces a special option `--fullJanno´ is not inconsistent in my view. A CLI is structured very differently than a URL, as it allows more expressiveness and hence can make things more explicit.

File: src/Poseidon/CLI/List.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (stschiff) [reply src/Poseidon/CLI/List.hs]: Done.

**Comment by nevrome** [comment]
As stated in the meeting: You have convinced me :+1:

**Pull request merged**